src/node/setupRemoteServer.ts

```diff
@@ -0,0 +1,212 @@
+import * as http from 'http'
+import { invariant } from 'outvariant'
+import { Server as WebSocketServer } from 'socket.io'
+import type { Socket } from 'socket.io-client'
+import { Emitter } from 'strict-event-emitter'
+import { DeferredPromise } from '@open-draft/deferred-promise'
+import {
+  LifeCycleEventsMap,
+  RequestHandler,
+  SetupApi,
+  handleRequest,
+  rest,
+} from '~/core'
+import {
+  SerializedRequest,
+  SerializedResponse,
+  deserializeRequest,
+  deserializeResponse,
+  serializeRequest,
+  serializeResponse,
+} from '~/core/utils/request/serializeUtils'
+import { LifeCycleEventEmitter } from '~/core/sharedOptions'
+
+export const SYNC_SERVER_URL = new URL('http://localhost:50222')
```

Comment:
I wonder if we can rely on a fixed port to always be available. I'm tempted to say yes but we should probably handle the scenario when this part is busy.